Try running `./python.exe Lib/test/test_genericalias.py`. There are three errors that occur without this too, but this one's new:
```

======================================================================
FAIL: test_collections_protocols_allowed (__main__.ProtocolTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/guido/pep585/Lib/test/test_typing.py", line 1381, in test_collections_protocols_allowed
    self.assertIsSubclass(B, Custom)
  File "/Users/guido/pep585/Lib/test/test_typing.py", line 40, in assertIsSubclass
    raise self.failureException(message)
AssertionError: <class '__main__.ProtocolTests.test_collections_protocols_allowed.<locals>.B'> is not a subclass of <class '__main__.ProtocolTests.test_collections_protocols_allowed.<locals>.Custom'>
```

--- a/Lib/_collections_abc.py
+++ b/Lib/_collections_abc.py
@@ -8,6 +8,7 @@ Unit tests are in test_collections.
 
 from abc import ABCMeta, abstractmethod
 import sys
+from types import GenericAlias
 
 __all__ = ["Awaitable", "Coroutine",
            "AsyncIterable", "AsyncIterator", "AsyncGenerator",
@@ -65,10 +66,6 @@ async def _ag(): yield
 _ag = _ag()
 async_generator = type(_ag)
 del _ag
-
-
-# needed for making various ABCs generic
-GenericAlias = type(list[str])
 
 
 ### ONE-TRICK PONIES ###

--- a/Lib/typing.py
+++ b/Lib/typing.py
@@ -1083,6 +1083,10 @@ class Protocol(Generic, metaclass=_ProtocolMeta):
 
             # Second, perform the actual structural compatibility check.
             for attr in _get_protocol_attrs(cls):
+                # Requiring people to implement this on e.g. iterable classes
+                # would break existing code
+                if attr == '__class_getitem__':
+                    continue
                 for base in other.__mro__:
                     # Check if the members appears in the class dictionary...
                     if attr in base.__dict__:

--- a/Lib/typing.py
+++ b/Lib/typing.py
@@ -946,7 +946,7 @@ _TYPING_INTERNALS = ['__parameters__', '__orig_bases__',  '__orig_class__',
 
 _SPECIAL_NAMES = ['__abstractmethods__', '__annotations__', '__dict__', '__doc__',
                   '__init__', '__module__', '__new__', '__slots__',
-                  '__subclasshook__', '__weakref__']
+                  '__subclasshook__', '__weakref__', '__class_getitem__']
 
 # These special attributes will be not collected as protocol members.
 EXCLUDED_ATTRIBUTES = _TYPING_INTERNALS + _SPECIAL_NAMES + ['_MutableMapping__marker']
@@ -1083,10 +1083,6 @@ class Protocol(Generic, metaclass=_ProtocolMeta):
 
             # Second, perform the actual structural compatibility check.
             for attr in _get_protocol_attrs(cls):
-                # Requiring people to implement this on e.g. iterable classes
-                # would break existing code
-                if attr == '__class_getitem__':
-                    continue
                 for base in other.__mro__:
                     # Check if the members appears in the class dictionary...
                     if attr in base.__dict__:

